docs/StringManifesto.md

```diff
@@ -1124,6 +1124,55 @@ collection of `UInt32`, where values less than `0x80` represent the
 corresponding ASCII character, and other values represent data that is specific
 to the underlying encoding of the string.
 
+### Unsafe String API
+
+While the goal of the String rewrite is to achieve good performance for most usage scenarios without jeopardizing Unicode compliance or type and memory safety, we recognize the importance of being able to squeeze the last bit of performance by dropping down to unsafe APIs.
+
+The following extension makes it possible to access and mutate the content of the String viewed as a contiguous buffer. 
+
+```swift
+extension String {
+    /// The base address of the continuous buffer containing the String elements
+    /// if the String is backed by a contiguous buffer.
+    ///
+    /// There is no guarantee that the String is backed by a contiguous buffer.
+    /// Even if it is, there is no guarantee that the buffer is writable.
+    public var baseAddress: UnsafeRawPointer? { get }
```

Comment:
Sure it could be. I decided to mimic the `StringBuffer` code (which predates `UnsafeRawBufferPointer`). My main motivation for not going for an `UnsafeRawBufferPointer` is the possible mismatch between the buffer size measured in bytes and measured in elements. I would expect the user to derive either an `UnsafeBufferPointer<UInt8>` (or `CChar` or raw) or an `UnsafeBufferPointer<UInt16>` from the base address. I am concerned that baking in the choice of width into the type even indirectly as an `UnsafeRawBufferPointer` may be a source of confusion.